application/org.openjdk.jmc.flightrecorder.flameview/src/main/java/org/openjdk/jmc/flightrecorder/flameview/views/FlameGraphView.java

```diff
@@ -155,7 +153,8 @@
 				jsFlameviewColoring);
 	}
 
-	private static final ExecutorService MODEL_EXECUTOR = Executors.newFixedThreadPool(1);
+	private static final int MODEL_EXECUTOR_THREADS_NUMBER = 3;
+	private static final ExecutorService MODEL_EXECUTOR = Executors.newFixedThreadPool(MODEL_EXECUTOR_THREADS_NUMBER);
```

Comment:
Perhaps add a thread factory to make it easier to interpret recordings?

	private static final ExecutorService MODEL_EXECUTOR = Executors.newFixedThreadPool(MODEL_EXECUTOR_THREADS_NUMBER, new ThreadFactory() {
		private ThreadGroup threadGroup = new ThreadGroup("FlameGraphModelCalculatorGroup");
		private AtomicInteger threadCounter = new AtomicInteger();
		@Override
		public Thread newThread(Runnable r) {
			Thread t = new Thread(threadGroup, r, "FlameGraphModelCalculator-" + threadCounter.getAndIncrement());
			t.setDaemon(true);
			return t;
		}
	});

Reply:
yeap, I will look better in threads list, added!